mesonbuild/ast/interpreter.py

```diff
@@ -85,11 +87,27 @@ class MockRunTarget(MesonInterpreterObject):
 _V = T.TypeVar('_V')
 
 
+@dataclasses.dataclass
+class GlobalState(_GlobalState):
```

Comment:
This has two different classes whose names are both `GlobalState`. This is just calling for them to get them mixed up.